nomad/structs/errors.go

```diff
@@ -78,6 +78,10 @@ func IsErrUnknownMethod(err error) bool {
 	return err != nil && strings.Contains(err.Error(), errUnknownMethod)
 }
 
+func IsErrRPCCoded(err error) bool {
+	return err != nil && strings.Contains(err.Error(), errRPCCodedErrorPrefix)
```

Comment:
`CodeFromRPCCodedErr` uses `strings.HasPrefix` - I would suggest having them be consistent.  Having `CodeFromRPCCodedErr` handle wrapping . and added prefixes would be nice.